tests/components/lifx/test_sensor.py

```diff
@@ -0,0 +1,196 @@
+"""Tests for the lifx integration sensor platform."""
+
+from homeassistant.components import lifx
+from homeassistant.components.lifx import DOMAIN, LIFXCoordination
+from homeassistant.const import (
+    ATTR_UNIT_OF_MEASUREMENT,
+    CONF_HOST,
+    SIGNAL_STRENGTH_DECIBELS,
+    TIME_MINUTES,
+    TIME_SECONDS,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.entity_component import async_update_entity
+from homeassistant.helpers.entity_registry import async_entries_for_config_entry
+from homeassistant.setup import async_setup_component
+
+from . import (
+    IP_ADDRESS,
+    MAC_ADDRESS,
+    MockFailingLifxCommand,
+    _mocked_bulb,
+    _mocked_hev_bulb,
+    _patch_config_flow_try_connect,
+    _patch_device,
+    _patch_discovery,
+)
+
+from tests.common import MockConfigEntry
+
+
+async def test_rssi_and_uptime_sensors(hass: HomeAssistant) -> None:
+    """Test the RSSI sensor on a LIFX bulb."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_HOST: IP_ADDRESS},
+        unique_id=MAC_ADDRESS,
+    )
+    config_entry.add_to_hass(hass)
+    bulb = _mocked_bulb()
+    with _patch_discovery(device=bulb), _patch_config_flow_try_connect(
+        device=bulb
+    ), _patch_device(device=bulb):
+        await async_setup_component(hass, lifx.DOMAIN, {lifx.DOMAIN: {}})
+        await hass.async_block_till_done()
+
+    entity_ids = ["sensor.my_bulb_rssi", "sensor.my_bulb_uptime"]
+
+    entity_registry = er.async_get(hass)
+    registry_entries = [
+        entity_registry.entities.get(entity_id) for entity_id in entity_ids
+    ]
+    assert [registry_entry.disabled for registry_entry in registry_entries]
+    assert [
+        registry_entry.disabled_by is er.RegistryEntryDisabler.INTEGRATION
+        for registry_entry in registry_entries
+    ]
+
+    enabled_entities = [
+        entity_registry.async_update_entity(entity_id, disabled_by=None)
+        for entity_id in entity_ids
+    ]
+    assert [not enabled_entity.disabled for enabled_entity in enabled_entities]
+
+    with _patch_discovery(device=bulb), _patch_config_flow_try_connect(
+        device=bulb
+    ), _patch_device(device=bulb):
+        await hass.config_entries.async_reload(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    states = [hass.states.get(entity_id) for entity_id in entity_ids]
+    assert states[0].state == "16"
+    assert states[1].state == "609815"
+    assert (
+        states[0].attributes.get(ATTR_UNIT_OF_MEASUREMENT) == SIGNAL_STRENGTH_DECIBELS
+    )
+    assert states[1].attributes.get(ATTR_UNIT_OF_MEASUREMENT) == TIME_SECONDS
+
+    disabled_entities = [
+        entity_registry.async_update_entity(
+            entity_id, disabled_by=er.RegistryEntryDisabler.INTEGRATION
+        )
+        for entity_id in entity_ids
+    ]
```

Comment:
This could be a helper

Reply:
As above, any chance you can point me towards an example?

Reply:
Traveling this week but should have some time later in the week after jet lag recovery